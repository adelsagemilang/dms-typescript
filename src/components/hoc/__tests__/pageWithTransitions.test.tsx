import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import PageWithTransitions from '../pageWithTransitions';

describe('<PageWithTransitions/>', () => {
  const TestWrappedComponent = () => <div className="test">Test</div>;
  const PageWithTransitionsComponent = PageWithTransitions(
    TestWrappedComponent,
    'test',
  );

  it('renders correctyly', () => {
    const wrapper = shallow(<PageWithTransitionsComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('animated width on 100ms', () => {
    const animateTime = sinon.useFakeTimers();
    const wrapper = shallow(<PageWithTransitionsComponent />);
    wrapper.update();
    animateTime.tick(100);
    expect(wrapper.state('isTransitioned')).toBe(true);
    expect(wrapper.hasClass('is-transitioned')).toBe(true);
  });
});
