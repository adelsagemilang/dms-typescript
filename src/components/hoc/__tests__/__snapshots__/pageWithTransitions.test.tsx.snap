// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWithTransitions/> renders correctyly 1`] = `
<div
  className="has-background-white has-transition-width "
>
  <withHelpersModifiers(Container)
    isFluid={true}
  >
    <HeadingWithButtons
      chartLink="/test"
      space={true}
      title="test"
    />
  </withHelpersModifiers(Container)>
  <TestWrappedComponent />
</div>
`;
